actors/abi/sector.go

```diff
@@ -179,50 +201,39 @@ type OnChainSealVerifyInfo struct {
 /// PoSting
 ///
 
-type ChallengeTicketsCommitment []byte
 type PoStRandomness Randomness
-type PartialTicket []byte // 32 bytes
-
-type PoStVerifyInfo struct {
-	Randomness      PoStRandomness
-	Candidates      []PoStCandidate // From OnChain*PoStVerifyInfo
-	Proofs          []PoStProof
-	EligibleSectors []SectorInfo
-	Prover          ActorID // used to derive 32-byte prover ID
-	ChallengeCount  uint64
-}
 
+// Information about a sector necessary for PoSt verification.
 type SectorInfo struct {
 	RegisteredProof // RegisteredProof used when sealing - needs to be mapped to PoSt registered proof when used to verify a PoSt
 	SectorNumber    SectorNumber
 	SealedCID       cid.Cid // CommR
 }
 
-type OnChainElectionPoStVerifyInfo struct {
-	Candidates []PoStCandidate // each PoStCandidate has its own RegisteredProof
-	Proofs     []PoStProof     // each PoStProof has its own RegisteredProof
-	Randomness PoStRandomness
-}
-
-type OnChainPoStVerifyInfo struct {
-	Candidates []PoStCandidate // each PoStCandidate has its own RegisteredProof
-	Proofs     []PoStProof     // each PoStProof has its own RegisteredProof
+type PoStProof struct {
+	RegisteredProof
+	ProofBytes []byte
 }
 
-type PoStCandidate struct {
-	RegisteredProof
-	PartialTicket  PartialTicket             // Optional —  will eventually be omitted for SurprisePoSt verification, needed for now.
-	PrivateProof   PrivatePoStCandidateProof // Optional — should be ommitted for verification.
-	SectorID       SectorID
-	ChallengeIndex int64
+// Information needed to verify a Winning PoSt attached to a block header.
+// Note: this is not used within the state machine, but by the consensus/election mechanisms.
+type WinningPoStVerifyInfo struct {
+	Randomness        PoStRandomness
+	Proofs            []PoStProof
+	ChallengedSectors []SectorInfo
+	EligibleSectors   []SectorNumber
```

Comment:
As discussed, this will grow to be a very large list. Two options instead of this include:
- use a bitfield of EligibleSectors instead, to be indexed inside proof verification
- split proof verification into two: generate the challenge index from randomness, and then verify with only the challenged sector infos

Reply:
it seems like this list will be at most ~2k sectors?

Reply:
No, for _winning_ PoSt it's all the sectors. 10s of millions.